sdk/src/azure/core/az_json_reader.c

```diff
@@ -700,6 +698,8 @@ AZ_NODISCARD static az_result _az_json_reader_process_number(az_json_reader* ref
   return AZ_OK;
 }
 
+AZ_INLINE int32_t _az_min(int32_t a, int32_t b) { return a < b ? a : b; }
```

Comment:
When do we decide to go with one over the other?

Reply:
do you mean an inline function vs a macro?  I have the same question.

Reply:
We should avoid macros, unless it is absolutely necessary, so it is impossible what we are trying to do using the normal functionality provided by the language. If it is not stated in the guidelines, we should put it there. Macros are harder to debug (try to step in), they would calculate argument twice (for instance, if `a` is a function call, it will be called twice when the macro decides to return `a` as result).

Reply:
An advantage of a macro would be if it was used heavily, and used with different types - longs, ints, doubles, etc. C doesn't have templates, so it would not be possible to write a function that accepts all types universally while having the same name (assuming that would be a benefit to outweigh all the cons of macros). But here, macro is only used in a single place.

Reply:
Maybe make the method static?

Reply:
`AZ_INLINE` is already static - it is define as `static inline`. We absolutely should've used static if it wasn't already.

Reply:
Ha! Guidelines even have `MAX()` macro as example, I didn't know/remember that :)